commit 38765c62b63a6357a0737fcd813a77d5934bcbc6 (refs/changes/65/2065/1)
Author: James Nugent <james@joyent.com>
Date:   2017-06-08T19:04:24-05:00 (2 years, 4 months ago)
    
    OS-6174 Use first NIC for admin network if admin_nic_autoselect=true
    
    This commit adds a configuration key to /usbkey/config with the effect
    of using the first NIC reported by `dladm show-phys` if no NIC has the
    tag `admin`.
    
    This is useful in environments such as Vagrant where the NIC
    configuration changes underneath you between OS 'installation' and the
    time of use.

diff --git a/overlay/generic/lib/svc/method/net-physical b/overlay/generic/lib/svc/method/net-physical
index 7fbae783..683360d0 100644
--- a/overlay/generic/lib/svc/method/net-physical
+++ b/overlay/generic/lib/svc/method/net-physical
@@ -498,6 +498,18 @@ if smf_is_globalzone; then
     setup_mtu
 
     # Setup admin NIC
+
+    # If there is no NIC with the admin tag, and the config has
+    # admin_nic_autoselect=true, designate the first NIC reported
+    # by dladm for admin use. This is useful in environments where
+    # the NICs are known to change beneath us.
+    if [[ "${CONFIG_admin_nic_autoselect}" == "true" ]] \
+        && ! nictagadm exists admin ; then
+        nictagadm add admin "$(dladm show-phys -m -p -o address | head -n1)"
+        nictagadm list
+        SYSINFO_NIC_admin=$(dladm show-phys -m -p -o link | head -n1)
+    fi
+
     if [[ -z "${SYSINFO_NIC_admin}" ]]; then
         echo "ERROR: admin NIC not found, unable to bring up admin network."
         exit ${SMF_EXIT_ERR_CONFIG}
